jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -273,26 +272,25 @@ public String getManagerHostPort() {
   }
 
   public static class ReporterConfiguration {
+    private static final int DEFAULT_FLUSH_INTERVAL_MS = 1000;
+    private static final int DEFAULT_MAX_QUEUE_SIZE = 100;
 
-    private static final String defaultAgentHost = "localhost";
-    private static final int defaultAgentPort = 5775;
-    private static final int defaultFlushIntervalMs = 1000;
-    private static final int defaultMaxQueueSize = 100;
-
-    private final Boolean logSpans;
-
-    private final String agentHost;
-
-    private final Integer agentPort;
-
-    private final Integer flushIntervalMs;
+    private Boolean logSpans;
+    private String agentHost;
+    private Integer agentPort;
+    private Integer flushIntervalMs;
+    private Integer maxQueueSize;
 
-    private final Integer maxQueueSize;
+    private Sender sender;
 
     public ReporterConfiguration() {
       this(null, null, null, null, null);
     }
 
+    public ReporterConfiguration(Sender sender) {
```

Comment:
@black-adder here is the constructor, if there is no need to create reporter with http sender from env props it does not make sense to add string type. We should clearly document Jaeger API, so users know what parameters to use.